### PR TITLE
Retire headers.go

### LINKED ISSUE #274: Remove headers.go
### Issue To Be Solved

This came up on recent PR #266 

Do we really need headers in emulator? @janezpodhostnik @turbolent 

I think it is only used for cadence `getBlock` functions, as we don't fork chain in emulator, returning block from storage seems to suffice. But I maybe missing something

in blocks.go maybe we can do something like:

```
func (b *blocks) ByHeightFrom(height uint64, header *flowgo.Header) (*flowgo.Header, error) {
       // bluesign: maybe some height check here needed 

       block, err := b.blockchain.storage.BlockByHeight(context.Background(), height)
	if err != nil {
		return nil, err
	}
	return block.Header, nil
}
```

I think this will make emulator - flow-go coupling looser.

### PR DESCRIPTION
**Pull request by bluesign**
Closes #274 

Discussion is in the referenced issue. 

______

For contributor use:

- [X] Targeted PR against `master` branch
- [X] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the GitHub PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: blocks.go [hunk 1]
```diff
@@ -68,6 +66,13 @@ func (b *blocks) FinalizedHeader() (*flowgo.Header, error) {
 }
 
 func (b *blocks) ByHeightFrom(height uint64, header *flowgo.Header) (*flowgo.Header, error) {
-	return environment.NewBlockFinder(b.headers).
-		ByHeightFrom(height, header)
+	if height > header.Height {
+		return nil, storage.ErrNotFound
+	}
```
Comment (janezpodhostnik) [inline blocks.go:71]: I think you can also add

```go
if header.Height == height {
    return header, nil
}
```

**Review by sideninja — APPROVED** [review]

**Review by janezpodhostnik — COMMENTED** [review]
Looks good!

Maybe just add a comment mentioning that we don't have to do anything complex, because the emulator doesn't have forks.

File: blocks.go (hunk 1 above)
Comment (bluesign) [reply blocks.go:71]: I think this one is in flow-go , but I will verify

Comment (janezpodhostnik) [reply blocks.go:72]: ah, didn't see that. Good catch

**Review by janezpodhostnik — APPROVED** [review]

**Comment by bluesign** [comment]
I messed up the branch when rebasing I guess, fixing

**Comment by janezpodhostnik** [comment]
I think you might need to rebase it to the top of master.

**Comment by bluesign** [comment]
@janezpodhostnik I think somehow GitHub showing wrong diff. ( I rebased once again )

**Comment by bluesign** [comment]
ok seems I managed to fix the problem somehow, I think can be merged.

**Pull request merged**